WordPress/gutenberg
# Add global styles sidebar at edit site screen

Previous attempts: https://github.com/WordPress/gutenberg/pull/20061 https://github.com/WordPress/gutenberg/pull/20530

This PR resumes the work to add a mechanism for the user to tweak the styles that are defined by core and theme via `experimental-theme.json`.

![2007-28-2119-global-styles-edit-site](https://user-images.githubusercontent.com/583546/88711364-5e50df00-d118-11ea-905d-e3aa6a8b3431.gif)

## Test instructions

Test in `edit-site`:

- Install and activate the [Global Styles demo theme](https://github.com/nosolosw/global-styles-theme/) (or any other with support for global styles). 
- Enable the FSE Experiment under "Gutenberg > Experiments".
- Go to "Gutenberg > Site Editor (beta)" and play with the Global Styles controls in the sidebar. For example, set the link color in the global tab.
- Save the changes to the Global Styles template via the "Update design" button in the editor.
- Go to the front and see that your changes are applied and are the same as the editor.

Test in `edit-post`:

- Install and activate the Global Styles demo theme.
- Enable the FSE Experiment under "Gutenberg > Experiments".
- Go to the post editor and create some content. Make sure the style changes done in edit-site are honored. For example, add two paragraphs with a link, verify the link color is the same than in `edit-site`.
- Select one of the paragraphs and edit its link color. Verify that the link color of that paragraph is updated but the other one remains the same.
